WordPressLint/src/main/java/org/wordpress/android/lint/MissingNullAnnotationDetector.kt

```diff
@@ -0,0 +1,179 @@
+package org.wordpress.android.lint
+
+import com.android.tools.lint.client.api.UElementHandler
+import com.android.tools.lint.detector.api.Category.Companion.CORRECTNESS
+import com.android.tools.lint.detector.api.Detector
+import com.android.tools.lint.detector.api.Implementation
+import com.android.tools.lint.detector.api.Issue
+import com.android.tools.lint.detector.api.JavaContext
+import com.android.tools.lint.detector.api.LintFix
+import com.android.tools.lint.detector.api.Scope.Companion.JAVA_FILE_SCOPE
+import com.android.tools.lint.detector.api.Severity
+import com.android.tools.lint.detector.api.SourceCodeScanner
+import com.android.tools.lint.detector.api.isJava
+import com.intellij.psi.PsiPrimitiveType
+import org.jetbrains.uast.UAnnotated
+import org.jetbrains.uast.UAnnotationMethod
+import org.jetbrains.uast.UElement
+import org.jetbrains.uast.UEnumConstant
+import org.jetbrains.uast.UField
+import org.jetbrains.uast.UMethod
+import org.jetbrains.uast.UParameter
+import org.jetbrains.uast.UVariable
+
+class MissingNullAnnotationDetector : Detector(), SourceCodeScanner {
+    override fun getApplicableUastTypes(): List<Class<out UElement>> = listOf(
+            UField::class.java,
+            UMethod::class.java,
+    )
+
+    override fun createUastHandler(context: JavaContext): UElementHandler? = with(context) {
+        if (!isJava(uastFile?.sourcePsi)) {
+            return null
+        }
+        object : UElementHandler() {
+            override fun visitField(node: UField) {
+                if (node.requiresNullAnnotation && !node.isNullAnnotated) {
+                    report(node, MISSING_FIELD_ANNOTATION)
+                }
+            }
+
+            override fun visitMethod(node: UMethod) {
+                node.uastParameters.forEach { visitParameter(node, it) }
+
+                if (node.requiresNullAnnotation && !node.isNullAnnotated) {
+                    report(node, MISSING_METHOD_RETURN_TYPE_ANNOTATION)
+                }
+            }
+
+            private fun visitParameter(node: UMethod, parameter: UParameter) {
+                if (parameter.requiresNullAnnotation && !parameter.isNullAnnotated) {
+                    if (node.isConstructor)
```

Comment:
Minor (🔍): Consider using braces when an `if/else` statement is multiline.

Reply:
Good idea. Resolved in https://github.com/wordpress-mobile/WordPress-Lint-Android/pull/11/commits/0bad70743f66680d143d1060b273192a369ae5d7

Reply:
Awesome, thanks for this update @mkevins ! 💯